pkg/providers/v1/aws.go

```diff
@@ -1302,19 +1312,9 @@ func newAWSCloud(cfg CloudConfig, awsServices Services) (*Cloud, error) {
 		return nil, fmt.Errorf("error creating AWS metadata client: %q", err)
 	}
 
-	err = updateConfigZone(&cfg, metadata)
-	if err != nil {
-		return nil, fmt.Errorf("unable to determine AWS zone from cloud provider config or EC2 instance metadata: %v", err)
-	}
-
-	zone := cfg.Global.Zone
-	if len(zone) <= 1 {
-		return nil, fmt.Errorf("invalid AWS zone in config file: %s", zone)
-	}
-	regionName, err := azToRegion(zone)
-	if err != nil {
-		return nil, err
-	}
+	regiondetails, err := getRegionFromMetadata(cfg, metadata)
```

Comment:
must check for err here

Reply:
I think the lack of err check causes TestNewAWSCloud to fail now after the refactor, try the test locally as well to verify .. https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-aws/313/pull-cloud-provider-aws-test/1499235237555605504